Create memory, timing, and weak scaling plots for STM paper

aeslaughter opened the pull request:
(refs #16137)

aeslaughter committed:
Create memory, timing, and weak scaling plots for STM paper

(refs #16137)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/753688/) on 4d974dd wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/18005/site)

This comment will be updated on new commits.

moosebuild commented:
All [jobs](https://civet.inl.gov/event/68917/) on a612bf6 : invalidated by @aeslaughter

loganharbour commented:
Status?

aeslaughter commented:
Need to get back to this. The funding for the effort dried up, but I should just finish the paper anyway.

moosebuild commented:
Job [Modules debug PETSc alt](https://civet.inl.gov/job/753761/) on a612bf6 : invalidated by @GiudGiud

GiudGiud commented:
Thanks a lot for this script @aeslaughter, it was great to look at scalability studies in MaCaw. If people look at this further, I added in MaCaw support for
- strong & weak scaling
- mpi and shared memory
- presplitting of meshes before distributed runs
- general support for all postprocessor plotting, easier than pre-built lists
and will add more generality to what curves are being shown on the same plot

aeslaughter committed:
Restrict to module only

This is required by the python script, which looks for the module executable not combined

(refs #16137)

aeslaughter committed:
Re-ran scaling studies for paper

aeslaughter committed:
Write results by default with --run and disable for test

(refs #16137)

moosebuild commented:
Job [Generate and verify coverage](https://civet.inl.gov/job/827072/) on 1e02a62 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>b79d3c</th>
      <th colspan=4>#18005 1e02a6</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>81.56%</td>
      <td>81.56%</td>
      <td>-</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>69452</td>
      <td>69452</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15706</td>
      <td>15706</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/18005/coverage_diff/framework.html)

## Modules coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>b79d3c</th>
      <th colspan=4>#18005 1e02a6</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>87.05%</td>
      <td>87.05%</td>
      <td>-</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>91072</td>
      <td>91072</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>13543</td>
      <td>13543</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/18005/coverage_diff/modules.html)

## Full coverage reports

<details><summary>Reports</summary>

- [framework](https://mooseframework.inl.gov/docs/PRs/18005/coverage/framework)
- [chemical_reactions](https://mooseframework.inl.gov/docs/PRs/18005/coverage/chemical_reactions)
- [combined](https://mooseframework.inl.gov/docs/PRs/18005/coverage/combined)
- [contact](https://mooseframework.inl.gov/docs/PRs/18005/coverage/contact)
- [external_petsc_solver](https://mooseframework.inl.gov/docs/PRs/18005/coverage/external_petsc_solver)
- [fluid_properties](https://mooseframework.inl.gov/docs/PRs/18005/coverage/fluid_properties)
- [fsi](https://mooseframework.inl.gov/docs/PRs/18005/coverage/fsi)
- [functional_expansion_tools](https://mooseframework.inl.gov/docs/PRs/18005/coverage/functional_expansion_tools)
- [geochemistry](https://mooseframework.inl.gov/docs/PRs/18005/coverage/geochemistry)
- [heat_conduction](https://mooseframework.inl.gov/docs/PRs/18005/coverage/heat_conduction)
- [level_set](https://mooseframework.inl.gov/docs/PRs/18005/coverage/level_set)
- [misc](https://mooseframework.inl.gov/docs/PRs/18005/coverage/misc)
- [navier_stokes](https://mooseframework.inl.gov/docs/PRs/18005/coverage/navier_stokes)
- [peridynamics](https://mooseframework.inl.gov/docs/PRs/18005/coverage/peridynamics)
- [phase_field](https://mooseframework.inl.gov/docs/PRs/18005/coverage/phase_field)
- [porous_flow](https://mooseframework.inl.gov/docs/PRs/18005/coverage/porous_flow)
- [ray_tracing](https://mooseframework.inl.gov/docs/PRs/18005/coverage/ray_tracing)
- [rdg](https://mooseframework.inl.gov/docs/PRs/18005/coverage/rdg)
- [richards](https://mooseframework.inl.gov/docs/PRs/18005/coverage/richards)
- [stochastic_tools](https://mooseframework.inl.gov/docs/PRs/18005/coverage/stochastic_tools)
- [tensor_mechanics](https://mooseframework.inl.gov/docs/PRs/18005/coverage/tensor_mechanics)
- [xfem](https://mooseframework.inl.gov/docs/PRs/18005/coverage/xfem)

</details>


This comment will be updated on new commits.
